pkg/controllers/subnet/subnet_controller.go

```diff
@@ -70,8 +73,41 @@ func (r *SubnetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		log.Error(err, "Unable to fetch Subnet CR", "req", req.NamespacedName)
 		return ResultRequeue, err
 	}
+
+	bindingCRs := r.getSubnetBindingCRsBySubnet(ctx, subnetCR)
+	if len(bindingCRs) > 0 {
+		if !controllerutil.ContainsFinalizer(subnetCR, servicecommon.SubnetFinalizerName) {
+			controllerutil.AddFinalizer(subnetCR, servicecommon.SubnetFinalizerName)
+			if err := r.Client.Update(ctx, subnetCR); err != nil {
+				log.Error(err, "Failed to add the finalizer", "Subnet", req.NamespacedName)
+				msgFailAddFinalizer := fmt.Sprintf("Failed to add the finalizer on a Subnet for the reference by SubnetConnectionBindingMap %s", bindingCRs[0].Name)
+				r.StatusUpdater.UpdateFail(ctx, subnetCR, err, "Failed to add the finalizer on Subnet used by SubnetConnectionBindingMaps", setSubnetReadyStatusFalse, msgFailAddFinalizer)
+				return ResultRequeue, err
+			}
+		}
```

Comment:
I left a comment before, but it's hard to find the tread.
You should skip this part when `!subnetCR.DeletionTimestamp.IsZero()` is true. 
Per my understanding, we can't stop users to delete a subnet CR. As long as the user run kubectl delete subnet $cr, the subnetCR.DeletionTimestamp.IsZero() will be non-zero. And you may still have bindingMap CRs associated to it. Then this part will be executed to add a finalizer to a deleted CR which doesn't make sense.